packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -1523,10 +2010,130 @@ fn _run(
                     }
                 }
             }
+
             Ok(())
         }.in_current_span();
     };
 
     let local = tokio::task::LocalSet::new();
     local.block_on(&runtime, impl_future)
 }
+
+/// Helper function to create a [v8::String]
+fn new_js_string<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    s: impl AsRef<str>,
+) -> JSRunnerResult<v8::Local<'s, v8::String>> {
+    let s = s.as_ref();
+    v8::String::new(scope, s)
+        .ok_or_else(|| JSRunnerError::V8(format!("Could not create String: {s}")))
+}
+
+/// C representation of Arrow array data nodes
+#[repr(C)]
+#[derive(Debug)]
+pub struct DataFfi<'s> {
+    pub len: usize,
+    pub null_count: usize,
+    pub n_buffers: u32,
+    pub buffer_ptrs: [*const u8; 2],
+    pub buffer_capacities: [usize; 2],
+    pub null_bits_ptr: *const u8,
+    pub null_bits_capacity: usize,
+    _phantom: std::marker::PhantomData<v8::Local<'s, ()>>,
+}
+
+/// Translation of the old mv8::mv8_data_node_from_js from C++ to Rust
+fn data_node_from_js<'s>(
```

Comment:
I see a lot of code duplication in this function. Can we avoid some by moving them out to a separate function?

Reply:
I added a `get_obj_property` helper function https://github.com/hashintel/hash/pull/414/commits/02af15786b564ada4f9992194281ff0a3281afb9